lib/specs.js

```diff
@@ -463,7 +463,9 @@ class Reader {
           delete schema.type; // Always overwrite the type.
           utils.copyOwnProperties(ref, schema);
         }
-        return Object.keys(schema).length > 1 ? schema : schema.type;
+        const isOptional = this._tk.next({id: 'operator', val: '?', silent: true});
+        const result = Object.keys(schema).length > 1 ? schema : schema.type;
+        return isOptional ? ['null', result]: result;
```

Comment:
The position of the `null` may not be right here: it should be the second element if the field has a non-null default value. See https://avro.apache.org/docs/1.11.1/idl-language/#unions for context. You could address this by moving this logic to `_readField` (arguably better also since optionality is a field-level concept rather than type) and using the default's value (if any) to determine the right position.

Reply:
Oh, I missed that, thank you!
Fixed my code accordingly :)